homeassistant/components/insteon/__init__.py

```diff
@@ -324,7 +327,15 @@ def add_all_link(service):
         """Add an INSTEON All-Link between two devices."""
         group = service.data.get(SRV_ALL_LINK_GROUP)
         mode = service.data.get(SRV_ALL_LINK_MODE)
-        link_mode = 1 if mode.lower() == SRV_CONTROLLER else 0
+        if mode.lower() == SRV_CONTROLLER:
```

Comment:
Please move the lowercase modification to the service schema. Use `vol.Lower`.

Reply:
I'm not quite sure what you mean here. The schema entries themselves are already lowercase (lines 89-91 in the PR). Are you saying to take the results of service.data.get and lowercase that with voluptuous.utils.Lower()?

Since the schema has already validated the contents of mode, can I just remove the calls to str.lower() entirely, or is voluptuous not case-sensitive? Forgive me, it's not a module I've worked with before.

Reply:
Yeah, if the mode string is already lowercase we don't need to make it lowercase after the `vol.In` check. We could add `vol.Lower` before this check to allow users to be less specific with the casing, but it's not strictly necessary.